DiscountDiscover/View Controllers/LoginViewController.m

```diff
@@ -36,13 +36,15 @@ - (IBAction)loginUser:(id)sender {
     NSString *password = self.passwordField.text;
     
     __weak typeof(self) weakSelf = self;
-    [PFUser logInWithUsernameInBackground:username password:password block:^(PFUser * user, NSError *  error) {
-        if (error != nil) {
-            [weakSelf showErrorAlert:@"Login Error" message:error.localizedDescription];
-        } else {
-            [weakSelf performSegueWithIdentifier:@"loginSegue" sender:nil];
-        }
-    }];
+    dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
The `logInWithUsernameInBackground:...` method doesn't need to be called on the main thread, the `showErrorAlert` and `performSegueWithIdentifier` methods do. So the `dispatch_async` line should be inside the `logInWithUsernameInBackground:` completion block.